apps/nova/src/pages/UpcomingLessons.tsx

```diff
@@ -37,6 +96,36 @@ const UpcomingLessons: React.FC = () => {
         hasMore={lessons.query.hasNextPage && !lessons.query.isPending}
         refetch={lessons.query.refetch}
       />
+      <RatingDialog
+        open={rateDialogInfo.enabled}
+        loading={tutorQuery.isLoading || ratingQuery.isLoading}
```

Comment:
The loading button should only be for when you are creating a new ratings not when you are fethcing the data. 


```suggestion
        loading={ratingQuery.isLoading}
```

@mostafakamar2308 suggested that we should use `isPending` ([github](https://github.com/TanStack/query/discussions/6297)).